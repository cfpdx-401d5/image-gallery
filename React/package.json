{
  "name": "image-gallery",
  "version": "0.1.0",
  "private": true,
  "dependencies": {
    "body-parser": "^1.17.1",
    "cors": "^2.8.1",
    "express": "^4.15.2",
    "mongoose": "^4.8.7",
    "morgan": "^1.8.1",
    "react": "^15.4.2",
    "react-dom": "^15.4.2"
  },
  "devDependencies": {
    "chai": "^3.5.0",
    "chai-http": "^3.0.0",
    "enzyme": "^2.7.1",
    "enzyme-to-json": "^1.5.0",
    "eslint": "^3.17.1",
    "eslint-plugin-react": "^6.10.0",
    "mocha": "^3.2.0",
    "nodemon": "^1.11.0",
    "react-addons-test-utils": "^15.4.2",
    "react-scripts": "0.9.4"
  },
  "scripts": {
    "start": "react-scripts start",
    "start-server": "nodemon server.js",
    "build": "react-scripts build",
    "test": "react-scripts test --env=jsdom",
    "eject": "react-scripts eject"
  }
}
