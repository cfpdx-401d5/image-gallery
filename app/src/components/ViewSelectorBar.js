import React, { PropTypes } from 'react';

function ViewSelectorBar(props) {
  return (
    <div className='view-select'>
      <button onClick={() => props.clickHandler('detail')}> Detail View </button>
      <button onClick={() => props.clickHandler('thumbnail')}> Thumbnail View </button>
      <button onClick={() => props.clickHandler('gallery')}> Gallery View </button>
      <br />
    </div>
  ); 
};

ViewSelectorBar.propTypes = {
  clickHandler: PropTypes.func.isRequired,
};

export default ViewSelectorBar;
