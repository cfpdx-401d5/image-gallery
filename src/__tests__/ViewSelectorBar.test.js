import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';

import ViewSelectorBar from '../ViewSelectorBar';

function testClickHandler(e) {
  return e;
}

describe('ViewSelectorBar component: ', function () {

  it('renders the ViewSelectorBar component', () => {
    const rendered = shallow(<ViewSelectorBar clickHandler={testClickHandler} />);
    expect(toJson(rendered)).toMatchSnapshot();
  });

});
