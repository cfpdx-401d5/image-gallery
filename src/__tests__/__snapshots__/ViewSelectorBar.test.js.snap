exports[`ViewSelectorBar component:  renders the ViewSelectorBar component 1`] = `
<div
  className="view-select">
  <button
    onClick={[Function]}>
     Detail View 
  </button>
  <button
    onClick={[Function]}>
     Thumbnail View 
  </button>
  <button
    onClick={[Function]}>
     Gallery View 
  </button>
  <br />
</div>
`;
